pkgs/development/tools/poac/default.nix

```diff
@@ -0,0 +1,125 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, cpm
+, git
+, cacert
+, boost179
+, fmt_8
+, icu
+, libarchive
+, libgit2
+, lz4
+, openssl
+, spdlog
+}:
+
+let
+  git2-cpp = fetchFromGitHub {
+    owner = "ken-matsui";
+    repo = "git2-cpp";
+    rev = "v0.1.0-alpha.1";
+    sha256 = "sha256-Ub0wrBK5oMfWGv+kpq/W1PN3yzpcfg+XWRFF/lV9VCY=";
+  };
+
+  glob = fetchFromGitHub {
+    owner = "p-ranav";
+    repo = "glob";
+    rev = "v0.0.1";
+    sha256 = "sha256-2y+a7YFBiYX8wbwCCWw1Cm+SFoXGB3ZxLPi/QdZhcdw=";
+  };
+
+  packageProjectCMake = fetchFromGitHub {
+    owner = "TheLartians";
+    repo = "PackageProject.cmake";
+    rev = "v1.3";
+    sha256 = "sha256-ZktftDrPo+JhBt0XKJekv0cyxIagvcgMcXZOBd4RtKs=";
+  };
+
+  mitamaCppResult = fetchFromGitHub {
+    owner = "LoliGothick";
+    repo = "mitama-cpp-result";
+    rev = "v9.3.0";
+    sha256 = "sha256-CWYVPpmPIZZTsqXKh+Ft3SlQ4C9yjUof1mJ8Acn5kmM=";
+  };
+
+  ninja = fetchFromGitHub {
+    owner = "ninja-build";
+    repo = "ninja";
+    rev = "1d4034f0ac2b5cfb809b5ab983d47c3cb2c78415";
+    sha256 = "sha256-ZRZ7YJG7LE5hHkwX31dDl+FNkW3J4DLYueqWmoqg+VQ=";
+  };
+
+  structopt = fetchFromGitHub {
+    owner = "p-ranav";
+    repo = "structopt";
+    rev = "e9722d3c2b52cf751ebc1911b93d9649c4e365cc";
+    sha256 = "sha256-jIfKUyY2QQ2/donywwlz65PY8u7xODGoG6SlNtUhwkg=";
+  };
+
+  toml11 = fetchFromGitHub {
+    owner = "ToruNiina";
+    repo = "toml11";
+    rev = "9086b1114f39a8fb10d08ca704771c2f9f247d02";
+    sha256 = "sha256-fHUElHO4ckNQq7Q88GdbHGxfaAvWoWtGB0eD9y2MnLo=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "poac";
+  version = "0.3.8";
+
+  src = fetchFromGitHub {
+    owner = "poacpm";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-NgFNiIg5XC3bkujN3lxET4btZRC5MltHQZdn/TSKLno=";
+  };
+
+  preConfigure = ''
+      mkdir -p ${placeholder "out"}/share/cpm
+      cp ${cpm}/share/cpm/CPM.cmake ${placeholder "out"}/share/cpm/CPM_0.35.1.cmake
```

Comment:
This is a bit confusing to me.

`CPM.cmake` is a file to be employed during the building, or after it?

Because if it should be present only during the build of `poac`, then it should not be put on `$out`.
On the other hand, if it should be present after it, inside the installed package, then this is not a `postConfigure` phase but a `postInstall` one.

Reply:
@AndersonTorres 

`CPM.cmake` is only needed when the configuration with CMake.
During build with make, installation, and after them, `CPM.cmake` is no longer needed.

Could you please tell me where it should be?